src/lib/ledger_catchup/ledger_catchup.ml

```diff
@@ -13,20 +13,33 @@ module Make (Inputs : Inputs.S) :
    and type state_hash := State_hash.t
    and type network := Inputs.Network.t = struct
   open Inputs
-  module Worker = Catchup_worker.Make (Inputs)
 
+  let fold_result_seq list ~init ~f =
+    Deferred.List.fold list ~init:(Ok init) ~f:(fun acc elem ->
+        match acc with
+        | Error e -> Deferred.return (Error e)
+        | Ok acc -> f acc elem )
+
+  (* We would like the async scheduler to context switch between each iteration 
+  of external transitions when trying to build breadcrumb_path. Therefore, this 
+  function needs to return a Deferred *)
   let construct_breadcrumb_path ~logger initial_staged_ledger
       external_transitions =
-    let open Or_error.Let_syntax in
+    let open Deferred.Or_error.Let_syntax in
     let%map _, breadcrumbs =
-      List.fold_result external_transitions ~init:(initial_staged_ledger, [])
-        ~f:(fun (staged_ledger, acc) transition_with_hash ->
-          let external_transition = With_hash.data transition_with_hash in
+      fold_result_seq external_transitions ~init:(initial_staged_ledger, [])
+        ~f:(fun (staged_ledger, acc) external_transition ->
           let diff =
             External_transition.staged_ledger_diff external_transition
           in
           let%map _, _, `Updated_staged_ledger staged_ledger =
-            Staged_ledger.apply ~logger staged_ledger diff
+            Deferred.return @@ Staged_ledger.apply ~logger staged_ledger diff
```

Comment:
This map is unnecessary. It doesn't hurt anything, but let's whiteboard so I can explain why it doesn't do what you think it does.